src/components/Table/GridTable.stories.tsx

```diff
@@ -1849,3 +1854,146 @@ export function Headers() {
+    evt.preventDefault();
+    evt.dataTransfer.clearData();
+    // console.log("onRowDragEnd", evt);
+    const spacerIndex = rows.findIndex((r) => r.id === spacerRow.id);
+    if (spacerIndex > 0) {
+      rows.splice(spacerIndex, 1);
+      setRows([...rows]);
+    }
+  };
+
+  const onDrop = (row: Data, data: DataType, evt: EventType) => {
+    evt.preventDefault();
+    evt.dataTransfer.clearData();
+
+    try {
+      const draggedRowData = JSON.parse(evt.dataTransfer.getData("text/plain")).row;
+
+      // make sure spacer is removed
+      const spacerIndex = rows.findIndex((r) => r.id === spacerRow.id);
+      const spacerOrder = rows[spacerIndex].order;
+      if (spacerIndex > 0) {
+        rows.splice(spacerIndex, 1);
+      }
+
+      // get rows in order (index matches order)
+      let orderedRows = rows.sort((a, b) => a.order - b.order);
+
+      // remove dragged row
+      const draggedRow = orderedRows.splice(draggedRowData.order, 1)[0];
+
+      // change the dragging row's order to ceil(spacer's order)
+      draggedRow.order = Math.ceil(spacerOrder);
+
+      // insert it at the index
+      orderedRows = [
+        ...orderedRows.slice(0, draggedRow.order),
+        draggedRow,
+        ...orderedRows.slice(draggedRow.order, orderedRows.length),
+      ];
+      // set row order to index
+      orderedRows.forEach((r, idx) => (r.order = idx));
+      setRows([...orderedRows]);
+    } catch {}
+  };
+
+  const onDragEnter = (row: Data, data: DataType, evt: EventType) => {
+    evt.preventDefault();
+    if (data.row.id === spacerRow.id) {
+      return;
+    }
+
+    const dir = evt.clientY > evt.currentTarget.offsetTop + 0.5 * evt.currentTarget.clientHeight ? 1 : -1;
+
+    // add a spacer above the row being dragged over
+    const spacerIndex = rows.findIndex((r) => r.id === spacerRow.id);
+    if (spacerIndex === -1) {
+      setRows([...rows, { ...spacerRow, order: data.row.order + dir * 0.1 }]);
+    } else {
+      rows[spacerIndex].order = data.row.order + dir * 0.1;
+      setRows([...rows]);
+    }
+  };
+
+  const onDragOver = (row: Data, data: DataType, evt: EventType) => {
+    evt.preventDefault();
+
+    if (data.row.id === spacerRow.id) {
+      return;
+    }
+
+    const spacerIndex = rows.findIndex((r) => r.id === spacerRow.id);
+    if (spacerIndex > 0) {
+      const dir = evt.clientY > evt.currentTarget.offsetTop + 0.5 * evt.currentTarget.clientHeight ? 1 : -1;
+
+      rows[spacerIndex].order = data.row.order + dir * 0.1;
+      setRows([...rows]);
+    }
+  };
+
+  return (
+    <GridTable
+      columns={[nameColumn, orderColumn, actionColumn]}
+      onRowDragStart={{ data: onDragStart, spacer: onDragStart }}
+      onRowDragEnd={{ data: onDragEnd, spacer: onDragEnd }}
+      onRowDrop={{ data: onDrop, spacer: onDrop }}
+      onRowDragEnter={{ data: onDragEnter, spacer: onDragEnter }}
+      onRowDragOver={{ data: onDragOver, spacer: onDragOver }}
```

Comment:
Thought: I think this makes a ton of sense for prototyping/spiking; wondering if maybe before shipping the final API might be more short & sweet, like something like:

* `onRowMove((row, oldIndex, newIndex) => ...)`

And have all of the low-level drag math, and ideally like management of that spacer row (being injected/moving around within the page's rows), all managed internally for the page... :thinking: 

We could also add a few props to our GridStyle, like `dragSpacerRow: { ...css stuff... }` to hopefully both have our out-of-the-box styles support it already, but also pages could tweak it if/as needed.